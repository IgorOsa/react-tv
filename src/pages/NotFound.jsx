import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import './NotFound.scss';

const NotFound = () => {
  const location = useLocation();

  return (
    <div className="error-page">
      <h1>
        404: Page
        {' '}
        {location.pathname}
        {' '}
        Not Found!
      </h1>
      <p>
        Back to
        {' '}
        <Link className="link-secondary" to="/">Home page</Link>
      </p>
    </div>
  );
};

export default NotFound;
