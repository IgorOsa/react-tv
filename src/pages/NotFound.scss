.error-page {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  text-align: center;
  height: calc(100vh - 490px - 40px);
}
