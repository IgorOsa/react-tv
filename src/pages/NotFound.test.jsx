import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import NotFound from './NotFound';

it('should renders NotFound page correctly', () => {
  const tree = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: '/someMockErrorlink' }]}>
      <NotFound />
    </MemoryRouter>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
