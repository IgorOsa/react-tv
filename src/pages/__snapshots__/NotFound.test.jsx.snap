// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders NotFound page correctly 1`] = `
<div
  className="error-page"
>
  <h1>
    404: Page
     
    /someMockErrorlink
     
    Not Found!
  </h1>
  <p>
    Back to
     
    <a
      className="link-secondary"
      href="/"
      onClick={[Function]}
    >
      Home page
    </a>
  </p>
</div>
`;
